extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -308,6 +308,14 @@ export class AmpA4A extends AMP.BaseElement {
      * @private {boolean}
      */
     this.isCollapsed_ = false;
+
+    // Note: This URL needs to be per-slot because we don't want ads in
+    // different slots to be able to interfere with each other, so they
+    // need to be cross-domain from each other.  However, we want the
+    // preconnect URL to be the same as the URL used in the fetch step
+    // (obviously), so generate it once here.
+    /** @const {string} */
+    this.nameframeUrl_ = getDefaultBootstrapBaseUrl(this.win, 'nameframe');
```

Comment:
Note that we do not reset the nameframeUrl_ as part of unlayoutCallback.  I think this is fine but wanted to point it out.